lib/conduit_mqtt/handler.ex

```diff
@@ -8,21 +8,31 @@ defmodule ConduitMQTT.Handler do
   import Conduit.Message
   require Logger
 
-  def init([client_id: _client_id, broker: _broker, name: _name, opts: _opts] = args) do
+  def init([client_id: _client_id, broker: _broker, name: _name, conn_type: _conn_type, opts: _opts] = args) do
```

Comment:
Create a state struct here, so you can only pattern match on the things you need to later.

``` elixir
def init(args) do
  {:ok, struct!(State, args)}
end
```

Reply:
You'll need to create the state module and call defstruct with the fields. You can do that in this file right before this function